editor/plugins/sprite_frames_editor_plugin.cpp

```diff
@@ -786,6 +787,44 @@ void SpriteFramesEditor::_animation_add() {
 	animations->grab_focus();
 }
 
+void SpriteFramesEditor::_animation_duplicate() {
+	if (!frames->has_animation(edited_anim)) {
+		return;
+	}
+
+	int counter = 1;
+	String name = String(edited_anim) + " " + itos(counter);
+	while (frames->has_animation(name)) {
+		counter++;
+		name = String(edited_anim) + " " + itos(counter);
+	}
+
+	undo_redo->create_action(TTR("Duplicate Animation"));
+	undo_redo->add_do_method(frames, "add_animation", name);
+	undo_redo->add_do_method(frames, "set_animation_loop", name, frames->get_animation_loop(edited_anim));
+	undo_redo->add_do_method(frames, "set_animation_speed", name, frames->get_animation_speed(edited_anim));
+	for (int i = 0; i < frames->get_frame_count(edited_anim); i++) {
+		undo_redo->add_do_method(frames, "add_frame", name, frames->get_frame(edited_anim, i));
+	}
+	undo_redo->add_undo_method(frames, "remove_animation", name);
+	undo_redo->add_do_method(this, "_update_library");
+	undo_redo->add_undo_method(this, "_update_library");
+
+	// Not sure whether this part is needed/wanted (`_animation_add` is doing the same thing).
```

Comment:
You should do something about this comment. Either use `_animation_add()` here if it's possible, or add a FIXME or something.